<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Figures from bulk RNA-seq analysis | Bioinformatics figures for publication</title>
  <meta name="description" content="This tutorial focuses on generating figures for publications from a bioinformatics analysis in answering Biological questions." />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Figures from bulk RNA-seq analysis | Bioinformatics figures for publication" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This tutorial focuses on generating figures for publications from a bioinformatics analysis in answering Biological questions." />
  <meta name="github-repo" content="chisangad/bioinf_figures" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Figures from bulk RNA-seq analysis | Bioinformatics figures for publication" />
  
  <meta name="twitter:description" content="This tutorial focuses on generating figures for publications from a bioinformatics analysis in answering Biological questions." />
  

<meta name="author" content="David Chisanga and Wei Shi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="prerequisites.html"/>
<link rel="next" href="analysis-protocol.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.onjcri.org.au/our-research/#bcgl">BCG Lab @ONJCRI</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#prereq"><i class="fa fa-check"></i><b>2.1</b> Data</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="prerequisites.html"><a href="prerequisites.html#bulk-rna-seq-data"><i class="fa fa-check"></i><b>2.1.1</b> Bulk RNA-seq data</a></li>
<li class="chapter" data-level="2.1.2" data-path="prerequisites.html"><a href="prerequisites.html#scrna-seq-data"><i class="fa fa-check"></i><b>2.1.2</b> scRNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#software"><i class="fa fa-check"></i><b>2.2</b> SOFTWARE</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figures-from-bulk-rna-seq-analysis.html"><a href="figures-from-bulk-rna-seq-analysis.html"><i class="fa fa-check"></i><b>3</b> Figures from bulk RNA-seq analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="figures-from-bulk-rna-seq-analysis.html"><a href="figures-from-bulk-rna-seq-analysis.html#background"><i class="fa fa-check"></i><b>3.1</b> Background</a></li>
<li class="chapter" data-level="3.2" data-path="figures-from-bulk-rna-seq-analysis.html"><a href="figures-from-bulk-rna-seq-analysis.html#multi-dimensional-scaling-plot-mds-or-pca"><i class="fa fa-check"></i><b>3.2</b> Multi-Dimensional scaling plot (MDS) or PCA</a></li>
<li class="chapter" data-level="3.3" data-path="figures-from-bulk-rna-seq-analysis.html"><a href="figures-from-bulk-rna-seq-analysis.html#mean-difference-plot"><i class="fa fa-check"></i><b>3.3</b> Mean-difference plot</a></li>
<li class="chapter" data-level="3.4" data-path="figures-from-bulk-rna-seq-analysis.html"><a href="figures-from-bulk-rna-seq-analysis.html#heatmaps"><i class="fa fa-check"></i><b>3.4</b> Heatmaps</a></li>
<li class="chapter" data-level="3.5" data-path="figures-from-bulk-rna-seq-analysis.html"><a href="figures-from-bulk-rna-seq-analysis.html#volcano-plots"><i class="fa fa-check"></i><b>3.5</b> Volcano plots</a></li>
<li class="chapter" data-level="3.6" data-path="figures-from-bulk-rna-seq-analysis.html"><a href="figures-from-bulk-rna-seq-analysis.html#pathway-analysis"><i class="fa fa-check"></i><b>3.6</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="figures-from-bulk-rna-seq-analysis.html"><a href="figures-from-bulk-rna-seq-analysis.html#gene-ontology-enrichment"><i class="fa fa-check"></i><b>3.6.1</b> Gene Ontology enrichment</a></li>
<li class="chapter" data-level="3.6.2" data-path="figures-from-bulk-rna-seq-analysis.html"><a href="figures-from-bulk-rna-seq-analysis.html#kegg-pathway-enrichment"><i class="fa fa-check"></i><b>3.6.2</b> KEGG pathway enrichment</a></li>
<li class="chapter" data-level="3.6.3" data-path="figures-from-bulk-rna-seq-analysis.html"><a href="figures-from-bulk-rna-seq-analysis.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>3.6.3</b> Gene Set Enrichment Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis-protocol.html"><a href="analysis-protocol.html"><i class="fa fa-check"></i><b>4</b> Analysis protocol</a>
<ul>
<li class="chapter" data-level="4.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#read-mapping-and-umi-counts"><i class="fa fa-check"></i><b>4.1</b> Read mapping and UMI counts</a></li>
<li class="chapter" data-level="4.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#quality-control-checks"><i class="fa fa-check"></i><b>4.2</b> Quality control checks</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#mitochondrial-content"><i class="fa fa-check"></i><b>4.2.1</b> Mitochondrial content</a></li>
<li class="chapter" data-level="4.2.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#ribosomal-content"><i class="fa fa-check"></i><b>4.2.2</b> Ribosomal content</a></li>
<li class="chapter" data-level="4.2.3" data-path="analysis-protocol.html"><a href="analysis-protocol.html#number-of-detected-genes"><i class="fa fa-check"></i><b>4.2.3</b> Number of detected genes</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="analysis-protocol.html"><a href="analysis-protocol.html#filtering"><i class="fa fa-check"></i><b>4.3</b> Filtering</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#cell-filtering"><i class="fa fa-check"></i><b>4.3.1</b> Cell filtering</a></li>
<li class="chapter" data-level="4.3.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#gene-filtering"><i class="fa fa-check"></i><b>4.3.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="analysis-protocol.html"><a href="analysis-protocol.html#normalization"><i class="fa fa-check"></i><b>4.4</b> Normalization</a></li>
<li class="chapter" data-level="4.5" data-path="analysis-protocol.html"><a href="analysis-protocol.html#dimension-reduction-and-clustering"><i class="fa fa-check"></i><b>4.5</b> Dimension reduction and clustering</a></li>
<li class="chapter" data-level="4.6" data-path="analysis-protocol.html"><a href="analysis-protocol.html#differential-expression-analysis-between-clusters"><i class="fa fa-check"></i><b>4.6</b> Differential expression analysis between clusters</a></li>
<li class="chapter" data-level="4.7" data-path="analysis-protocol.html"><a href="analysis-protocol.html#cell-annotation"><i class="fa fa-check"></i><b>4.7</b> Cell annotation</a></li>
<li class="chapter" data-level="4.8" data-path="analysis-protocol.html"><a href="analysis-protocol.html#trajectory-analysis"><i class="fa fa-check"></i><b>4.8</b> Trajectory analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a>
<ul>
<li class="chapter" data-level="4.9" data-path="references.html"><a href="references.html#r-markdown"><i class="fa fa-check"></i><b>4.9</b> R Markdown</a></li>
<li class="chapter" data-level="4.10" data-path="references.html"><a href="references.html#including-plots"><i class="fa fa-check"></i><b>4.10</b> Including Plots</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics figures for publication</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figures-from-bulk-rna-seq-analysis" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Figures from bulk RNA-seq analysis<a href="figures-from-bulk-rna-seq-analysis.html#figures-from-bulk-rna-seq-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="background" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Background<a href="figures-from-bulk-rna-seq-analysis.html#background" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Previous results from the <a href="https://chisangad.github.io/bulkRNAseqtut/index.html">Bulk RNA-seq analysis workshop</a> were saved in the “Workshop_RNAseq” directory on All staff share as an R-object “Counts2.RData” or can also be downloaded from <a href="">here</a>. Import this into R to run the rest of the analysis</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#load the required package</span></span>
<span id="cb1-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Counts2.RData&quot;</span>) <span class="co"># loads the counts dataset</span></span></code></pre></div>
<p>There are a number of ways in which you can visualize the results from RNA-seq, while in this chapter we attempt to cover the most common ways of visualizing the data, they are by no means exhaustive.</p>
</div>
<div id="multi-dimensional-scaling-plot-mds-or-pca" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Multi-Dimensional scaling plot (MDS) or PCA<a href="figures-from-bulk-rna-seq-analysis.html#multi-dimensional-scaling-plot-mds-or-pca" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Multi-Dimensional scaling (MDS) plot is a means of visualizing the level of similarity of individual samples in a dataset. In the plot, the distance between each pair of samples is the root-mean-square deviation (Euclidean distance) for the top genes. Distances on the plot can be interpreted as leading log2-fold-change, meaning the typical (root-mean-square) log2-fold-change between the samples for the genes that distinguish those samples.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb2-1" aria-hidden="true" tabindex="-1"></a>col<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;orange&quot;</span>,<span class="st">&quot;purple&quot;</span>)[<span class="fu">factor</span>(targets<span class="sc">$</span>CellType)]</span>
<span id="cb2-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(</span>
<span id="cb2-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb2-3" aria-hidden="true" tabindex="-1"></a>  y,</span>
<span id="cb2-4"><a href="figures-from-bulk-rna-seq-analysis.html#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex =</span> <span class="fl">0.8</span>,</span>
<span id="cb2-5"><a href="figures-from-bulk-rna-seq-analysis.html#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">col=</span>col</span>
<span id="cb2-6"><a href="figures-from-bulk-rna-seq-analysis.html#cb2-6" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-2"></span>
<img src="Bioinfo-figures_files/figure-html/unnamed-chunk-2-1.png" alt="MDS plot showing speration and similarity of samples" width="672" />
<p class="caption">
Figure 3.1: MDS plot showing speration and similarity of samples
</p>
</div>
</div>
<div id="mean-difference-plot" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Mean-difference plot<a href="figures-from-bulk-rna-seq-analysis.html#mean-difference-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A mean-difference plot or MD-plot is a plot that can be used to show the fold-change differences against the average expression values of all genes used in the analysis. This can be generated easily using the <em>plotMD</em> function within the <strong>limma</strong> package.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMD</span>(fit.contr, <span class="at">column =</span> <span class="dv">1</span>,<span class="at">status =</span> dt,<span class="at">cex=</span><span class="fl">0.8</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:figmd"></span>
<img src="Bioinfo-figures_files/figure-html/figmd-1.png" alt="Mean-Difference plot. Represents relationship between mean expression and fold-changes" width="576" />
<p class="caption">
Figure 3.2: Mean-Difference plot. Represents relationship between mean expression and fold-changes
</p>
</div>
</div>
<div id="heatmaps" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Heatmaps<a href="figures-from-bulk-rna-seq-analysis.html#heatmaps" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Heatmaps are a great way of demonstrating the differences in the expression patterns between different conditions in your dataset. Here we use the <em>coolmap</em> function within the <strong>limma</strong> package which is an extension of the <em>heatmap.2</em> function within the <strong>gplots</strong> package. The heatmap below shows the expression pattern of the top 100 differentially expressed genes between CIS vs WT.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb4-1" aria-hidden="true" tabindex="-1"></a>heat.data <span class="ot">&lt;-</span> y<span class="sc">$</span>E[<span class="fu">head</span>(<span class="fu">rownames</span>(de.genes), <span class="at">n =</span> <span class="dv">100</span>), ]</span>
<span id="cb4-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heat.data) <span class="ot">&lt;-</span> targets<span class="sc">$</span>Sample</span>
<span id="cb4-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heat.data) <span class="ot">&lt;-</span> <span class="fu">head</span>(de.genes<span class="sc">$</span>Symbol, <span class="at">n =</span> <span class="dv">100</span>)</span>
<span id="cb4-4"><a href="figures-from-bulk-rna-seq-analysis.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">coolmap</span>(</span>
<span id="cb4-5"><a href="figures-from-bulk-rna-seq-analysis.html#cb4-5" aria-hidden="true" tabindex="-1"></a>  heat.data,</span>
<span id="cb4-6"><a href="figures-from-bulk-rna-seq-analysis.html#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">margin =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>),</span>
<span id="cb4-7"><a href="figures-from-bulk-rna-seq-analysis.html#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">cexCol =</span> <span class="fl">0.8</span>,</span>
<span id="cb4-8"><a href="figures-from-bulk-rna-seq-analysis.html#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">lhei =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="dv">4</span>),</span>
<span id="cb4-9"><a href="figures-from-bulk-rna-seq-analysis.html#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">cexRow =</span> <span class="fl">0.6</span></span>
<span id="cb4-10"><a href="figures-from-bulk-rna-seq-analysis.html#cb4-10" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:figheatmap"></span>
<img src="Bioinfo-figures_files/figure-html/figheatmap-1.png" alt="Heatmap plot of differentially expressed genes in CIS vs WT" width="576" />
<p class="caption">
Figure 3.3: Heatmap plot of differentially expressed genes in CIS vs WT
</p>
</div>
</div>
<div id="volcano-plots" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Volcano plots<a href="figures-from-bulk-rna-seq-analysis.html#volcano-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A volcano plot shows the relationship between the log fold-change on the x-axis and a measure of statistical significance on the y-axis. The measure of significance can be -log(p-value) or the B-statistics. Here, we will use the <em>volcanoplot</em> function within the <strong>limma</strong> package.In the volcano plot below, differentially expressed genes are highlighted with red for up-regulated and blue for down-regulated genes.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#highlight de genes</span></span>
<span id="cb5-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb5-2" aria-hidden="true" tabindex="-1"></a>col<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>,<span class="st">&quot;black&quot;</span>,<span class="st">&quot;red&quot;</span>)[<span class="fu">factor</span>(dt[,<span class="st">&quot;CISvsWT&quot;</span>])]</span>
<span id="cb5-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">volcanoplot</span>(fit.contr,<span class="at">col=</span>col)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:figvolc"></span>
<img src="Bioinfo-figures_files/figure-html/figvolc-1.png" alt="Volcano plot. Shows the relationship between fold-change and significance of change" width="480" />
<p class="caption">
Figure 3.4: Volcano plot. Shows the relationship between fold-change and significance of change
</p>
</div>
<p>Alternatively, an interactive mean-difference plot can be generated using <em>glMDPlot</em> in <strong>Glimma</strong> which outputs an interactive html page with results displayed in the left panel and the expression values for the selected gene. A gene within the dataset can be searched through a search bar.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(Glimma))</span>
<span id="cb6-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb6-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb6-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb6-3" aria-hidden="true" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;Glimma&quot;</span>)</span>
<span id="cb6-4"><a href="figures-from-bulk-rna-seq-analysis.html#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(Glimma)</span>
<span id="cb6-5"><a href="figures-from-bulk-rna-seq-analysis.html#cb6-5" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Loading required package: Glimma</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">glMDPlot</span>(fit.contr, <span class="at">coef=</span><span class="dv">1</span>, <span class="at">status=</span>dt, <span class="at">main=</span><span class="fu">colnames</span>(fit.contr)[<span class="dv">1</span>],</span>
<span id="cb8-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb8-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">side.main=</span><span class="st">&quot;ENTREZID&quot;</span>, <span class="at">counts=</span>y<span class="sc">$</span>E, <span class="at">groups=</span><span class="fu">colnames</span>(contr),</span>
<span id="cb8-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb8-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">launch =</span> F)</span></code></pre></div>
<p><a href="glimma-plots/MD-Plot.html">Click here for interactive version</a></p>
<p>Hover over points to see sample-wise expression</p>
<ul>
<li><p>Click on column names to sort by column</p></li>
<li><p>Click rows on tables to highlight gene</p></li>
</ul>
</div>
<div id="pathway-analysis" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Pathway Analysis<a href="figures-from-bulk-rna-seq-analysis.html#pathway-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>From the differential expression analysis there were some 3181 differentially expressed genes. This is quiet a long list of genes to interpret and nearly impossible to go through a gene at a time to gain any meaningful biological understanding. A common downstream analysis step is to understand pathways/gene networks that the genes are implicated in.</p>
<div id="gene-ontology-enrichment" class="section level3 hasAnchor" number="3.6.1">
<h3><span class="header-section-number">3.6.1</span> Gene Ontology enrichment<a href="figures-from-bulk-rna-seq-analysis.html#gene-ontology-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Gene ontology (<a href="http://www.geneontology.org/" class="uri">http://www.geneontology.org/</a>) provides a controlled vocabulary for describing biological processes (BP ontology), molecular functions (MF ontology) and cellular components (CC ontology).</p>
<p>The GO ontologies themselves are organism-independent; terms are associated with genes for a specific organism through direct experimentation or through sequence homology with another organism and its GO annotation.</p>
<p>Terms are related to other terms through parent-child relationships in a directed acylic graph.</p>
<p>You can use enrichment analysis as another way of drawing conclusions from your set of differentially expressed genes.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb9-1" aria-hidden="true" tabindex="-1"></a>enrich.pvalue<span class="ot">&lt;-</span><span class="fl">0.00001</span></span></code></pre></div>
<p>Here, we use the <em>goana</em> function within the <strong>limma</strong> package to test for enrichment of differentially expressed genes between the CIS vs WT. Note that the <em>p-values</em> returned by <em>goana</em> are <strong>unadjusted for multiple testing</strong>. It is therfore, advisable that if the results are to be published, only terms with very small p-values should be included. For instance, in the example below, only terms with a p-value &lt; 10^{-5} are retained.</p>
<p>We use the output from the differential expression analysis and test for overrepresentation for the Up and Down regulated genes separately.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;GO.db&quot;</span>))</span>
<span id="cb10-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb10-2" aria-hidden="true" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GO.db&quot;</span>)</span>
<span id="cb10-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb10-4"><a href="figures-from-bulk-rna-seq-analysis.html#cb10-4" aria-hidden="true" tabindex="-1"></a>go.rst<span class="ot">&lt;-</span><span class="fu">goana</span>(fit.contr,<span class="at">species=</span><span class="st">&quot;Mm&quot;</span>)</span>
<span id="cb10-5"><a href="figures-from-bulk-rna-seq-analysis.html#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">#exclude terms not meeting our cut-off</span></span>
<span id="cb10-6"><a href="figures-from-bulk-rna-seq-analysis.html#cb10-6" aria-hidden="true" tabindex="-1"></a>go.rst<span class="ot">&lt;-</span>go.rst[<span class="fu">rowSums</span>(go.rst[,<span class="fu">c</span>(<span class="st">&quot;P.Up&quot;</span>,<span class="st">&quot;P.Down&quot;</span>)]<span class="sc">&lt;</span>enrich.pvalue)<span class="sc">==</span><span class="dv">1</span>,]</span>
<span id="cb10-7"><a href="figures-from-bulk-rna-seq-analysis.html#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="fu">topGO</span>(go.rst)</span></code></pre></div>
<pre><code>##                                                                      Term Ont
## GO:0050794                                 regulation of cellular process  BP
## GO:0050789                               regulation of biological process  BP
## GO:0065007                                          biological regulation  BP
## GO:0048522                        positive regulation of cellular process  BP
## GO:0048518                      positive regulation of biological process  BP
## GO:0060255                  regulation of macromolecule metabolic process  BP
## GO:0032501                               multicellular organismal process  BP
## GO:0080090                        regulation of primary metabolic process  BP
## GO:0031323                       regulation of cellular metabolic process  BP
## GO:0019222                                regulation of metabolic process  BP
## GO:0048856                               anatomical structure development  BP
## GO:0051171              regulation of nitrogen compound metabolic process  BP
## GO:0007275                             multicellular organism development  BP
## GO:0048731                                             system development  BP
## GO:0010604         positive regulation of macromolecule metabolic process  BP
## GO:0032502                                          developmental process  BP
## GO:0016043                                cellular component organization  BP
## GO:0019219 regulation of nucleobase-containing compound metabolic process  BP
## GO:0005488                                                        binding  MF
## GO:0005515                                                protein binding  MF
##               N  Up Down      P.Up       P.Down
## GO:0050794 6226 613 1214 1.0000000 3.968162e-76
## GO:0050789 6496 645 1244 1.0000000 5.133210e-74
## GO:0065007 6800 693 1263 1.0000000 1.837622e-65
## GO:0048522 3596 349  787 1.0000000 4.457436e-56
## GO:0048518 3848 382  815 1.0000000 5.304817e-52
## GO:0060255 3923 362  815 1.0000000 5.609124e-48
## GO:0032501 3664 351  774 1.0000000 1.387305e-47
## GO:0080090 3719 335  781 1.0000000 4.594288e-47
## GO:0031323 3836 358  799 1.0000000 4.641674e-47
## GO:0019222 4239 405  860 1.0000000 4.840800e-47
## GO:0048856 3188 303  697 1.0000000 6.835712e-47
## GO:0051171 3613 325  764 1.0000000 7.170605e-47
## GO:0007275 2778 264  628 1.0000000 4.847006e-46
## GO:0048731 2365 220  558 1.0000000 9.262732e-46
## GO:0010604 2233 219  535 0.9999990 1.255960e-45
## GO:0032502 3437 328  732 1.0000000 2.419765e-45
## GO:0016043 3877 406  798 0.9999999 1.281803e-44
## GO:0019219 2587 207  592 1.0000000 1.689384e-44
## GO:0005488 8062 894 1365 1.0000000 2.003374e-44
## GO:0005515 5698 597 1061 1.0000000 2.922417e-44</code></pre>
<p>A simple way to visualise the enrichment results is through a bar plot as shown below which shows the top 10 enriched terms in each direction.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-1" aria-hidden="true" tabindex="-1"></a>top.up <span class="ot">&lt;-</span> <span class="fu">head</span>(go.rst[<span class="fu">order</span>(go.rst<span class="sc">$</span>P.Up, <span class="at">decreasing =</span> F),], <span class="at">n =</span> <span class="dv">10</span>)</span>
<span id="cb12-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-2" aria-hidden="true" tabindex="-1"></a>top.down <span class="ot">&lt;-</span></span>
<span id="cb12-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(go.rst[<span class="fu">order</span>(go.rst<span class="sc">$</span>P.Down, <span class="at">decreasing =</span> F),], <span class="at">n =</span> <span class="dv">10</span>)</span>
<span id="cb12-4"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-4" aria-hidden="true" tabindex="-1"></a>bar.data <span class="ot">&lt;-</span></span>
<span id="cb12-5"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(</span>
<span id="cb12-6"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(</span>
<span id="cb12-7"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">Term =</span> top.up<span class="sc">$</span>Term,</span>
<span id="cb12-8"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">P.value =</span> top.up<span class="sc">$</span>P.Up,</span>
<span id="cb12-9"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">Dir =</span> <span class="st">&quot;up&quot;</span></span>
<span id="cb12-10"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-10" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb12-11"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(</span>
<span id="cb12-12"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">Term =</span> top.down<span class="sc">$</span>Term,</span>
<span id="cb12-13"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">P.value =</span> top.down<span class="sc">$</span>P.Down,</span>
<span id="cb12-14"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">Dir =</span> <span class="st">&quot;down&quot;</span></span>
<span id="cb12-15"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-15" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb12-16"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-16" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb12-17"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-17" aria-hidden="true" tabindex="-1"></a>bar.data <span class="ot">&lt;-</span> bar.data[<span class="fu">order</span>(bar.data<span class="sc">$</span>P.value, <span class="at">decreasing =</span> F),]</span>
<span id="cb12-18"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mai =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="dv">2</span>, <span class="fl">0.5</span>, <span class="fl">0.5</span>))</span>
<span id="cb12-19"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-19" aria-hidden="true" tabindex="-1"></a>bb <span class="ot">&lt;-</span> <span class="fu">barplot</span>(</span>
<span id="cb12-20"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-20" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span><span class="fu">log10</span>(bar.data<span class="sc">$</span>P.value),</span>
<span id="cb12-21"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">horiz =</span> T,</span>
<span id="cb12-22"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">&quot; -log10(p-value)&quot;</span>,</span>
<span id="cb12-23"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.names =</span> <span class="fl">0.5</span>,</span>
<span id="cb12-24"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="fu">factor</span>(bar.data<span class="sc">$</span>Dir)]</span>
<span id="cb12-25"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-25" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-26"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-26" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(</span>
<span id="cb12-27"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-27" aria-hidden="true" tabindex="-1"></a>  <span class="dv">2</span>,</span>
<span id="cb12-28"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">line =</span> <span class="sc">-</span><span class="fl">0.8</span>,</span>
<span id="cb12-29"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">at =</span> bb,</span>
<span id="cb12-30"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">labels =</span> bar.data<span class="sc">$</span>Term,</span>
<span id="cb12-31"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">tick =</span> F,</span>
<span id="cb12-32"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">las =</span> <span class="dv">2</span>,</span>
<span id="cb12-33"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.axis =</span> <span class="fl">0.5</span></span>
<span id="cb12-34"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-34" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-35"><a href="figures-from-bulk-rna-seq-analysis.html#cb12-35" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>,<span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;Up&quot;</span>,<span class="st">&quot;Down&quot;</span>),<span class="at">pch =</span> <span class="dv">22</span>,<span class="at">pt.bg =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:enrichplot"></span>
<img src="Bioinfo-figures_files/figure-html/enrichplot-1.png" alt="Enrichment plot. Top enrichments in either direction" width="672" />
<p class="caption">
Figure 3.5: Enrichment plot. Top enrichments in either direction
</p>
</div>
<p><strong>Add gene information (optional)</strong></p>
<p>Sometimes you may want to know what genes are enriched in each of the terms, this section allows you to add gene details.</p>
<p>First, we extract the genes that are associated with the catalytic activity GO terms. This results in a list mapping the GO term and the associated genes.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(GO.db))</span>
<span id="cb13-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb13-2" aria-hidden="true" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GO.db&quot;</span>)</span>
<span id="cb13-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(org.Mm.eg.db))</span>
<span id="cb13-4"><a href="figures-from-bulk-rna-seq-analysis.html#cb13-4" aria-hidden="true" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;org.Mm.eg.db&quot;</span>)</span></code></pre></div>
<pre><code>## Loading required package: org.Mm.eg.db</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb15-1" aria-hidden="true" tabindex="-1"></a>go.EntrezID <span class="ot">&lt;-</span> <span class="fu">as.list</span>(org.Mm.egGO2ALLEGS)</span>
<span id="cb15-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb15-2" aria-hidden="true" tabindex="-1"></a>go.rst <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(go.rst, <span class="at">var =</span> <span class="st">&quot;GO.ID&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Add gene details</span></span>
<span id="cb16-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-2" aria-hidden="true" tabindex="-1"></a>go.rst<span class="sc">$</span>Genes.Up <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(go.rst<span class="sc">$</span>GO.ID, <span class="cf">function</span>(x) {</span>
<span id="cb16-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-3" aria-hidden="true" tabindex="-1"></a>  xx <span class="ot">&lt;-</span> go.EntrezID[[x]]</span>
<span id="cb16-4"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(xx) <span class="sc">|</span></span>
<span id="cb16-5"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">with</span>(go.rst[go.rst<span class="sc">$</span>GO.ID <span class="sc">==</span> x,], Up <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span></span>
<span id="cb16-6"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-6" aria-hidden="true" tabindex="-1"></a>           P.Up <span class="sc">&gt;</span> enrich.pvalue))</span>
<span id="cb16-7"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="st">&quot;-&quot;</span>)</span>
<span id="cb16-8"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-8" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">with</span>(de.genes,EntrezID[logFC<span class="sc">&gt;</span><span class="dv">0</span>]),xx)</span>
<span id="cb16-9"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-9" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span></span>
<span id="cb16-10"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="fu">with</span>(fit.contr<span class="sc">$</span>genes, Symbol[EntrezID <span class="sc">%in%</span> xx]), <span class="at">collapse =</span> <span class="st">&quot;|&quot;</span>)</span>
<span id="cb16-11"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(x)</span>
<span id="cb16-12"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-12" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb16-13"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-13" aria-hidden="true" tabindex="-1"></a>go.rst<span class="sc">$</span>Genes.Down <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(go.rst<span class="sc">$</span>GO.ID, <span class="cf">function</span>(x) {</span>
<span id="cb16-14"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-14" aria-hidden="true" tabindex="-1"></a>  xx <span class="ot">&lt;-</span> go.EntrezID[[x]]</span>
<span id="cb16-15"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(xx) <span class="sc">|</span></span>
<span id="cb16-16"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">with</span>(go.rst[go.rst<span class="sc">$</span>GO.ID <span class="sc">==</span> x,], Down <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span></span>
<span id="cb16-17"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-17" aria-hidden="true" tabindex="-1"></a>           P.Down <span class="sc">&gt;</span> enrich.pvalue))</span>
<span id="cb16-18"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="st">&quot;-&quot;</span>)</span>
<span id="cb16-19"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-19" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">with</span>(de.genes,EntrezID[logFC<span class="sc">&lt;</span><span class="dv">0</span>]),xx)</span>
<span id="cb16-20"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-20" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span></span>
<span id="cb16-21"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="fu">with</span>(fit.contr<span class="sc">$</span>genes, Symbol[EntrezID <span class="sc">%in%</span> xx]), <span class="at">collapse =</span> <span class="st">&quot;|&quot;</span>)</span>
<span id="cb16-22"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(x)</span>
<span id="cb16-23"><a href="figures-from-bulk-rna-seq-analysis.html#cb16-23" aria-hidden="true" tabindex="-1"></a>}))</span></code></pre></div>
</div>
<div id="kegg-pathway-enrichment" class="section level3 hasAnchor" number="3.6.2">
<h3><span class="header-section-number">3.6.2</span> KEGG pathway enrichment<a href="figures-from-bulk-rna-seq-analysis.html#kegg-pathway-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Pathway enrichment can also be performed against the Kyoto Encyclopedia of Genes and Genomes (KEGG) (<a href="https://www.genome.jp/kegg/" class="uri">https://www.genome.jp/kegg/</a>) database. The KEGG database contains curated molecular pathways and disease signatures. To perform this analysis we use the <em>kegga</em> function within the <strong>limma</strong> which is similar to the <em>goana</em> function we used above.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb17-1" aria-hidden="true" tabindex="-1"></a>kegg.rst<span class="ot">&lt;-</span><span class="fu">kegga</span>(fit.contr,<span class="at">species=</span><span class="st">&quot;Mm&quot;</span>)</span>
<span id="cb17-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">topKEGG</span>(kegg.rst)</span></code></pre></div>
<pre><code>##                                               Pathway    N  Up Down
## path:mmu03008       Ribosome biogenesis in eukaryotes   75  37    4
## path:mmu01100                      Metabolic pathways 1038 206   75
## path:mmu04810        Regulation of actin cytoskeleton  143   6   45
## path:mmu05206                     MicroRNAs in cancer  121   6   40
## path:mmu04919       Thyroid hormone signaling pathway   93   3   32
## path:mmu03010                                Ribosome  137  39    1
## path:mmu01240               Biosynthesis of cofactors  111  32    2
## path:mmu04330                 Notch signaling pathway   43   2   18
## path:mmu00310                      Lysine degradation   52   1   20
## path:mmu03020                          RNA polymerase   32  14    1
## path:mmu00190               Oxidative phosphorylation  118  32    0
## path:mmu04062             Chemokine signaling pathway  119   5   34
## path:mmu04611                     Platelet activation   77   0   25
## path:mmu05200                      Pathways in cancer  323  26   71
## path:mmu05205                 Proteoglycans in cancer  126   8   34
## path:mmu04658        Th1 and Th2 cell differentiation   69   6   22
## path:mmu05224                           Breast cancer   81   8   24
## path:mmu05202 Transcriptional misregulation in cancer  121  16   32
## path:mmu03050                              Proteasome   44  15    0
## path:mmu01232                   Nucleotide metabolism   65  19    0
##                       P.Up       P.Down
## path:mmu03008 1.182585e-14 9.951210e-01
## path:mmu01100 5.674601e-12 1.000000e+00
## path:mmu04810 9.998630e-01 4.679254e-08
## path:mmu05206 9.987434e-01 5.746746e-08
## path:mmu04919 9.996663e-01 4.319019e-07
## path:mmu03010 6.123733e-07 1.000000e+00
## path:mmu01240 4.561006e-06 9.999989e-01
## path:mmu04330 9.789811e-01 6.418221e-06
## path:mmu00310 9.991527e-01 9.325271e-06
## path:mmu03020 1.341355e-05 9.916924e-01
## path:mmu00190 1.832128e-05 1.000000e+00
## path:mmu04062 9.995796e-01 2.081620e-05
## path:mmu04611 1.000000e+00 2.435669e-05
## path:mmu05200 9.971966e-01 4.089375e-05
## path:mmu05205 9.933378e-01 7.494761e-05
## path:mmu04658 8.861202e-01 1.000116e-04
## path:mmu05224 8.230146e-01 1.784375e-04
## path:mmu05202 4.708773e-01 1.811724e-04
## path:mmu03050 2.086057e-04 1.000000e+00
## path:mmu01232 3.146027e-04 1.000000e+00</code></pre>
</div>
<div id="gene-set-enrichment-analysis" class="section level3 hasAnchor" number="3.6.3">
<h3><span class="header-section-number">3.6.3</span> Gene Set Enrichment Analysis<a href="figures-from-bulk-rna-seq-analysis.html#gene-set-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>GO and KEGG enrichment analysis uses the list of differentially expressed genes to test for over-representation of these genes. As such the results typically depend on the cut-offs used to get the differentially expressed genes.</p>
<p>Gene Set Enrichment Analysis is instead performed using <em>fry</em>, a gene set enrichment function within the <strong>limma</strong> package. Unlike above where we focused on a subset of genes(differentially genes), here all the genes in the set are used to test for gene expression signatures or pathways. A test is performed against the GO terms for demonstrative purposes, alternatively if you wish to test against a specific gene set, this can easily be done by importing that list.</p>
<p>We then use <em>fry</em> to test for enrichment between CIS and WT. We use the design matrix ‘design’,the expression values ‘y’ and the contrast matrix ‘contr’ from the imported results to perform the gene set test and the GO term and gene map that we created above.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb19-1" aria-hidden="true" tabindex="-1"></a>go.all<span class="ot">&lt;-</span>go.EntrezID[go.rst<span class="sc">$</span>GO.ID]</span>
<span id="cb19-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(go.all)<span class="ot">&lt;-</span>go.rst<span class="sc">$</span>Term</span>
<span id="cb19-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb19-3" aria-hidden="true" tabindex="-1"></a>fry.GSEA<span class="ot">&lt;-</span><span class="fu">fry</span>(y,<span class="at">index =</span> go.all,<span class="at">design =</span> design,<span class="at">contrast =</span> contr)</span>
<span id="cb19-4"><a href="figures-from-bulk-rna-seq-analysis.html#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fry.GSEA)</span></code></pre></div>
<pre><code>##                                NGenes Direction       PValue         FDR
## meiotic chromosome segregation     63      Down 6.596898e-06 0.005277518
## chromosomal region                301      Down 1.338511e-04 0.041588170
## meiotic cell cycle process        132      Down 1.559556e-04 0.041588170
## meiotic nuclear division          118      Down 3.266901e-04 0.052601704
## DNA repair                        469      Down 3.827861e-04 0.052601704
## 90S preribosome                    28        Up 5.065661e-04 0.052601704
##                                PValue.Mixed  FDR.Mixed
## meiotic chromosome segregation  0.007482120 0.03370157
## chromosomal region              0.017736925 0.03370157
## meiotic cell cycle process      0.011471864 0.03370157
## meiotic nuclear division        0.010804063 0.03370157
## DNA repair                      0.030554738 0.03370939
## 90S preribosome                 0.003922337 0.03370157</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb21-1" aria-hidden="true" tabindex="-1"></a>term.test<span class="ot">&lt;-</span><span class="fu">rownames</span>(fry.GSEA)[<span class="dv">1</span>]</span></code></pre></div>
<p>Next, we can visualise the gene set enrichment results using a barcode plot. The barcode plot is used to plot the positions of genes within a set in ranked list of statistics. The statistics are ranked from left to right ie. from smallest to largest.</p>
<p>To do so, the gene identifies for the term should first be mapped to the index position of each gene in the expression matrix;</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="figures-from-bulk-rna-seq-analysis.html#cb22-1" aria-hidden="true" tabindex="-1"></a>index<span class="ot">&lt;-</span><span class="fu">rownames</span>(fit.contr) <span class="sc">%in%</span> go.all[[<span class="fu">rownames</span>(fry.GSEA)[<span class="dv">1</span>]]]</span>
<span id="cb22-2"><a href="figures-from-bulk-rna-seq-analysis.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barcodeplot</span>(fit.contr<span class="sc">$</span>coefficients[,<span class="st">&quot;CISvsWT&quot;</span>],<span class="at">index =</span> index,<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;CS&quot;</span>,<span class="st">&quot;WT&quot;</span>),</span>
<span id="cb22-3"><a href="figures-from-bulk-rna-seq-analysis.html#cb22-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">main=</span><span class="fu">rownames</span>(fry.GSEA)[<span class="dv">1</span>])</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-11"></span>
<img src="Bioinfo-figures_files/figure-html/unnamed-chunk-11-1.png" alt="Barcode plot showing the enrichment of meiotic chromosome segregation in CS vs WT" width="672" />
<p class="caption">
Figure 3.6: Barcode plot showing the enrichment of meiotic chromosome segregation in CS vs WT
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="prerequisites.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="analysis-protocol.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-bulkRNAseq.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinfo-figures.pdf", "Bioinfo-figures.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
